Update mkdocs.yml

Brandoncharleskhoo1 opened the pull request:
Updated the name from Marketing cloud to Salesforce Marketing cloud

# Amplitude Developer Docs PR


## Description

Describe your changes. 

## Deadline

When do these changes need to be live on the site?


## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.


@amplitude-dev-docs

bgiori commented:
![Screen Shot 2023-04-21 at 6 47 34 PM](https://user-images.githubusercontent.com/3282199/233755504-701ff047-be22-483d-97e5-365c737c05fa.png)


1. can we change the nav title to "Salesforce Marketing V1" and "Salesforce Marketing V2" to  keep them from wrapping
2. can we change the titles of the docs to "Send Cohorts to Salesforce Marketing Cloud V1" and "Send Cohorts to Salesforce Marketing Cloud V2"

Brandoncharleskhoo1 commented:
Thanks for the feedback. Made the changes as per your suggestions
1. Changed the nav title to "Salesforce Marketing V1" and "Salesforce Marketing V2" to keep them from wrapping
2. Changed the the titles of the docs to "Send Cohorts to Salesforce Marketing Cloud V1" and "Send Cohorts to Salesforce Marketing Cloud V2"
